### ISSUE TITLE
Notation for non-traditional f-plane is obscure

### ISSUE THREAD

**Issue by glwagner** [climate-machine/Oceananigans.jl#616]
Right now the "non-traditional" f-plane struct uses `f` and `f'` to denote what I presume are the `z` and `y` components of the rotation axis:

https://github.com/climate-machine/Oceananigans.jl/blob/f080635b93bdab9da7d1189ed21f27523c0ff810/src/Coriolis/f_plane.jl#L59

I think a better notation would be

```julia
struct NonTraditionalFPlane{T, T}
    fz :: T
    fy :: T
end
```
This is more explicit and follows the standard convention that the components of a vector are appended with direction.

**Comment by glwagner** [comment]
The documentation is also incorrect as the rotation axis is not arbitrary. It must be in the y-z plane.

**Comment by glwagner** [comment]
The dynamics functions are not tested, either because the functionality is not used in time-stepping. This is also a problem for `BetaPlane`.

**Comment by suyashbire1** [comment]
I have a test case for beta-plane incoming. 

Is there a way to test `x_f_cross_U` et al?

**Comment by glwagner** [comment]
> Is there a way to test x_f_cross_U et al?

Yes: construct a model with the given feature and then performing a time-step.

I guess ideally we would form a giant matrix representing all possible model configurations and time-step them all once. This would be expensive.

Maybe we should just loop over each physics specification individually and cross our fingers that there's no bad interactions (I can already tell this would be naive, for example, when a turbulence closure depends on buoyancy).

**Comment by glwagner** [comment]
I'm going to close this, as the primary concern was addressed by #617. We can discuss better testing on another issue.
